app/dojo.py

```diff
@@ -0,0 +1,164 @@
+from os import sys, path
+
+from app.office import Office
+from app.livingspace import LivingSpace
+from app.room import Room
+from app.utilities import Utilities
+from app.staff import Staff
+from app.fellow import Fellow
+from app.file import File
+from app.person import Person
+from app.errors import WrongFormatException
+
+sys.path.append(path.dirname(path.dirname(
+    path.dirname(path.abspath(__file__)))))
+
+
+class Dojo(object):
+    """docstring for Dojo"""
+
+    @classmethod
+    def create_room(cls, room_type, room_names):
+        type_of_room = room_type.strip().lower()
+        if type_of_room == "office":
+            return Office.create_office(room_names)
+        elif type_of_room == "livingspace":
+            return LivingSpace.create_livingspace(room_names)
+        else:
+            print("Invalid type of room")
+
+    @classmethod
+    def add_person(cls, surname, firstname, category, wants_accomodation=None):
+        type_of_person = category.strip().lower()
+        try:
+            if type_of_person == "staff":
+                new_staff = Staff(surname, firstname)
+                if new_staff:
+                    print(("Staff {ns.surname} {ns.firstname} has been"
+                           + " successfully added").format(ns=new_staff))
+                    Staff.add_to_staff_list(new_staff)
+                    Person.add_to_map(new_staff)
+                    cls.allocate_room(new_staff)
+
+            elif type_of_person == "fellow":
+                new_fellow = Fellow(surname, firstname, True) \
+                    if wants_accomodation in ['y', 'Y'] \
+                    else Fellow(surname, firstname)
+                if new_fellow:
+                    print(("Fellow {nf.surname} {nf.firstname} has been"
+                           + " successfully added").format(nf=new_fellow))
+                    Fellow.add_to_fellow_list(new_fellow)
+                    Person.add_to_map(new_fellow)
+                    cls.allocate_room(new_fellow, wants_accomodation)
+            else:
+                print("Invalid type of person")
+
+        except WrongFormatException as e:
```

Comment:
the exception is catching exception that might emanate from line 35 or 44 when instantiating a staff or fellow. It checks if whether the surname or first name is not made up of special characters only. The exception is raised in the Person class

class Person():
 +	"""docstring for Person"""
 +
 +	id_map = {}
 +
 +	def __init__(self, lname, fname):
 +		try:
 +			Utilities.check_format_validity([lname, fname])
 +		except WrongFormatException:
 +			raise WrongFormatException(
 +				"Firstname or Lastname is not a valid name format")
 +
 +		self.id = Utilities.generate_person_id(Person.get_id_list())
 +		self.surname = lname.capitalize()
 +		self.firstname = fname.capitalize()
 +		self.office = None